src/content/structured/get-started/testing-with-react-testing-library.mdx

```diff
@@ -0,0 +1,231 @@
+
+## Shadow DOM Testing Library
+
+However, RTL does not provide utility functions which go beyond the `shadowDOM`. As `@ukic/react` are React wrapped web components, which use the shadowDOM, it is not possible to exclusively use RTL to create React tests. Fortunately, with the addition of [Shadow DOM Testing Library](https://github.com/KonnorRogers/shadow-dom-testing-library), it is possible to select elements in the shadowDOM and interact with them by using methods provided by RTL. Shadow DOM Testing Library provides functions with 'Shadow' prefixed prior to the query type (e.g. `getByShadowText()`).
+
+```tsx
+// DOM tree for IcAlert
+<ic-alert class="dark hydrated" message="Foo" role="alert">
+  <ShadowRoot>
+    ...
+    <ic-typography class="ic-typography-body hydrated">
+      <ShadowRoot>
+        <slot />
+      </ShadowRoot>
+      Foo
+    </ic-typography>
+    ...
+  </ShadowRoot>
+</ic-alert>
+```
+
+```tsx
+import { IcAlert } from "@ukic/react";
+import { render, waitFor } from "@testing-library/react";
+import { screen } from "shadow-dom-testing-library";
+
+it("renders IcAlert", async () => {
+  const { getByRole /* container */ } = render(<IcAlert message="Foo" />);
+
+  // Wait for IcAlert to hydrate before continuing to view the full IcAlert DOM tree.
+  await waitFor(() => {
+    expect(getByRole("alert")).toHaveClass("hydrated");
+
+    // container can also be used with querySelector to select ic-alert
+    // expect(container.querySelector('ic-alert')).toHaveClass('hydrated');
+  });
+
+  expect(await screen.findByShadowText("Foo")).toBeInTheDocument();
+});
+```
+
+## Examples with IcSelect and IcTextField
+
+Below are additional examples on testing `@ukic/react` components using RTL and Shadow DOM Testing Library using `IcSelect` and `IcTextField`.
+
+
+### IcSelect
+
+```jsx
+// Select.tsx
+import { useState } from 'react';
+import { IcSelect } from '@ukic/react';
+import { IcMenuOption } from '@ukic/web-components';
+
+interface IcMenuOptions {
+    options: IcMenuOption[]
+}
+
+const Select = ({ options }: IcMenuOptions) => {
+    const [selectedValue, setSelectedValue] = useState<string | undefined>(
+        undefined
+    );
+    return (
+        <IcSelect
+            label='Title'
+            onIcChange={(e) => setSelectedValue(e.detail.value)}
+            options={options}
+            value={selectedValue}
+        />
+    );
+};
+  
+export default Select;
+```
+
+```jsx
+// Select.test.tsx
+import { fireEvent, render, waitFor } from '@testing-library/react';
+import { screen } from 'shadow-dom-testing-library';
+import Select from '../components/Select';
+import userEvent from '@testing-library/user-event';
+
+const options = [
+  { label: 'Americano', value: 'americano' }
+  { label: 'Espresso', value: 'espresso' },
+]
+
+describe('IcSelect', () => {
```

Comment:
Could you make left indentations more consistent across the different code sections please?